src/Campaigns/Actions/RedirectLegacyCreateFormToCreateCampaign.php

```diff
@@ -33,30 +38,73 @@ public function __invoke()
      */
     private function isAddingNewForm(): bool
     {
-        global $pagenow;
+        return $GLOBALS['pagenow'] === 'post-new.php'
+               && isset($_GET['post_type'])
+               && $_GET['post_type'] === 'give_forms';
+    }
 
-        $isOptionBasedFormEditorPage = $pagenow === 'post-new.php';
-        $isVisualFormBuilderPage = $pagenow === 'edit.php' && isset($_GET['page']) && 'givewp-form-builder' === $_GET['page'];
-        $isGiveFormsCpt = isset($_GET['post_type']) && $_GET['post_type'] === 'give_forms';
+    /**
+     * @unreleased
+     */
+    private function isEditingCampaignForm(): bool
+    {
+        return $GLOBALS['pagenow'] === 'edit.php'
+               && isset($_GET['post_type'])
+               && $_GET['post_type'] === 'give_forms'
+               && isset($_GET['page']) && $_GET['page'] === 'givewp-form-builder';
+    }
 
-        return ($isOptionBasedFormEditorPage || $isVisualFormBuilderPage) && $isGiveFormsCpt;
+    /**
+     * @unreleased
+     */
+    private function isP2P(): bool
+    {
+        return isset($_GET['p2p']);
```

Comment:
Is there anyway we don't have to have this like can we just see if the form is associated to a campaign and if the type is p2p?

Reply:
As I said, P2P doesn't use the `campaign_type` column. 
It would be easy if we added p2p meta to the form, but that is not the case. We need something to identify where to form belongs, and I believe this is our only option.